assets/js/googlesitekit/data/create-error-store.test.js

```diff
@@ -0,0 +1,221 @@
+				const selectedError = select.getErrorForSelector( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				expect( selectedError ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getErrorForAction', () => {
+			it( 'requires a selector param', () => {
+				expect( () => {
+					select.getErrorForAction();
+				} ).toThrow( 'actionName is required.' );
+			} );
+
+			it( 'returns undefined when error does not exist', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const selectedError = select.getErrorForAction( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( selectedError ).toBeUndefined();
+			} );
+
+			it( 'returns the specified error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const selectedError = select.getErrorForAction( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				expect( selectedError ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getError', () => {
+			it( 'requires a baseName param', () => {
+				select.getError();
+
+				expect( store.getState().error ).toBeUndefined();
+			} );
+
+			it( 'returns the correct error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+
+				const error = select.getError( ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				expect( error ).toMatchObject( ERROR_SAMPLE_2.error );
+			} );
+		} );
+
+		describe( 'getErrors', () => {
+			it( 'returns an empty array if there are no errors', () => {
+				const errors = select.getErrors();
+				expect( errors ).toEqual( [] );
+			} );
+
+			it( 'checks if we have only 1 error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				const errors = select.getErrors();
+				expect( errors ).toHaveLength( 1 );
+			} );
+
+			it( 'checks if we have more than 1 error', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const errors = select.getErrors();
+				expect( errors ).toHaveLength( 2 );
+			} );
+
+			it( 'checks if duplicate errors are removed', () => {
+				dispatch.receiveError( ERROR_SAMPLE.error, ERROR_SAMPLE.baseName, ERROR_SAMPLE.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				dispatch.receiveError( ERROR_SAMPLE_2.error, ERROR_SAMPLE_2.baseName, ERROR_SAMPLE_2.args );
+				const errors = select.getErrors();
+				expect( errors ).toHaveLength( 2 );
+			} );
```

Comment:
This test actually doesn't test the functionality of the unique list handling because receiving the same error multiple times will only overwrite the same key in `state.errors` (it won't actually add it multiple times).

This is only relevant when receiving the _same error_ multiple times, with different args, like what we do for the legacy compatibility. You can see what I mean here: https://github.com/google/site-kit-wp/blob/f244c63925cd56d0bb03a048d0b387824c335fe5/assets/js/googlesitekit/data/create-fetch-store.js#L159-L165 

So the return here should be unique regardless of any `baseName`/`args` provided 🙂